### ISSUE TITLE
Unable to get the same result: inline @Query() type vs @ApiQuery({type: ...})

### ISSUE THREAD

**Issue by pbn4** [nestjs/swagger#1022]
## I'm submitting a...

<pre><code>
[ ] Regression 
[*] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
I have a controller method where I would like to specify ApiQuery type at runtime:

```
@Get()
@ApiQuery({name: '', type: MyDto})
async list(@Query() queryParams: any) {
  return ...
}
```

where MyDto:

```
class MyDto {
  @Expose()
  @ApiProperty()
  id!: string

  @Expose()
  @ApiProperty()
  otherKey!: string
}
```

instead of inlining it like this:

```
@Get()
async list(@Query() queryParams: MyDto) {
  return ...
}
```

but the former produces a different/incorrect result. Generated `/docs` page contains a query param which is a single object-like field whereas the latter nicely creates 'id' and 'otherKey' as separate query params. 

## Expected behavior
The outcome to be consistent between @ApiQuery and inlining type into method signature.

## Environment

<pre><code>
Nest version: latest

 
For Tooling issues:
- Node version: 15
- Platform:  MacOs

Others:
Might be related to: https://github.com/nestjs/swagger/issues/1013

**Comment by kamilmysliwiec** [comment]
Please provide a minimum reproduction repository.

**Comment by pbn4** [comment]
Sure. Please take a look at /docs and compare whats in the AppController. https://github.com/pbn4/nestjs-query-issue/blob/master/src/app.controller.ts

**Comment by kamilmysliwiec** [comment]
This should be fixed in 4.7.3

**Comment by pbn4** [comment]
Thank you
